src/Stratis.Bitcoin/P2P/Protocol/Behaviors/EnforcePeerVersionCheckBehavior.cs

```diff
@@ -0,0 +1,105 @@
+﻿using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using NBitcoin;
+using Stratis.Bitcoin.Configuration;
+using Stratis.Bitcoin.P2P.Peer;
+using Stratis.Bitcoin.Utilities;
+using TracerAttributes;
+
+namespace Stratis.Bitcoin.P2P.Protocol.Behaviors
+{
+    /// <summary>
+    /// Sets the minimum supported client version <see cref="this.NodeSettings.MinProtocolVersion"> to <see cref="this.Network.Consensus.Options.EnforcedMinProtocolVersion">
+    /// based on the predefined block height <see cref="this.Network.Consensus.Options.EnforceMinProtocolVersionAtBlockHeight">.
+    /// Once the new minimum supported client version is changed all existing peer connections will be dropped upon the first received message from outdated client.
+    /// </summary>
+    public class EnforcePeerVersionCheckBehavior : NetworkPeerBehavior
+    {
+        /// <summary>An indexer that provides methods to query the best chain (the chain that is validated by the full consensus rules)</summary>
+        protected ChainIndexer ChainIndexer { get; private set; }
+
+        /// <summary>User defined node settings.</summary>
+        protected NodeSettings NodeSettings { get; private set; }
+
+        /// <summary>Specification of the network the node runs on - regtest/testnet/mainnet.</summary>
+        protected Network Network { get; private set; }
+
+        /// <summary>Logger factory usded while cloning the object.</summary>
+        private ILoggerFactory loggerFactory;
```

Comment:
I believe this can be `private readonly`?

Reply:
Defect: loggerFactory is not assigned in the constructor